drivers/ieee802154/ieee802154_rf2xx.c

```diff
@@ -0,0 +1,893 @@
+		 *   2. CCA_ED_DONE: 8.5.4
+		 *      • Indicates the end of a CCA or ED
+		 *        measurement. 8.6.4
+		 *  IRQ_3 (TRX_END)
+		 *    RX: Indicates the completion of a frame
+		 *      reception. 7.1.3
+		 *    TX: Indicates the completion of a frame
+		 *      transmission. 7.1.3
+		 *  IRQ_2 (RX_START) Indicates the start of a PSDU
+		 *    reception; the AT86RF233 state changed to BUSY_RX;
+		 *    the PHR can be read from Frame Buffer. 7.1.3
+		 *  IRQ_1 (PLL_UNLOCK) Indicates PLL unlock. If the radio
+		 *    transceiver is in BUSY_TX / BUSY_TX_ARET state, the
+		 *    PA is turned off immediately. 9.7.5
+		 *  IRQ_0 (PLL_LOCK) Indicates PLL lock.
+		 */
+		if (isr_status & (1 << RX_START)) {
+			ctx->trx_state = RF2XX_TRX_PHY_BUSY_RX;
+		} else if (isr_status & (1 << TRX_END)) {
+			if (ctx->trx_state == RF2XX_TRX_PHY_BUSY_RX) {
+				/* Set PLL_ON to avoid transceiver receive
+				 * new data until finish reading process
+				 */
+				rf2xx_trx_set_state(dev,
+					RF2XX_TRX_PHY_STATE_CMD_PLL_ON);
+				rf2xx_trx_rx(dev);
+				rf2xx_trx_set_state(dev,
+					RF2XX_TRX_PHY_STATE_CMD_RX_AACK_ON);
+			/*if (ctx->trx_state == RF2XX_TRX_PHY_BUSY_TX)*/
+			} else {
+				ctx->trx_trac =
+					(rf2xx_iface_reg_read(dev,
+						TRX_STATE_REG) >>
+						TRAC_STATUS) & 7;
+				k_sem_give(&ctx->trx_tx_sync);
+				rf2xx_trx_set_rx_state(dev);
+			}
+			ctx->trx_state = RF2XX_TRX_PHY_STATE_IDLE;
+		}
+		k_mutex_unlock(&ctx->phy_mutex);
+	}
+}
+
+/********************
+ * Radio aux API *
+ *******************/
+static inline u8_t *get_mac(struct device *dev)
+{
+	struct rf2xx_context *ctx = dev->driver_data;
+
+	/*
+	 * The RF2XX has two 32-bit registers for the MAC address where
+	 * 40 bits of the registers are factory programmed to be unique
+	 * and the rest are to be assigned as the "company-specific" value.
+	 * 802.15.4 defines a EUI-64 64-bit address with company specific
+	 * being 24 or 36 bits with the unique value being 24 or 40 bits.
+	 */
+
+	u32_t *ptr = (u32_t *)(ctx->mac_addr);
+
+	UNALIGNED_PUT(sys_rand32_get(), ptr);
+	ptr = (u32_t *)(ctx->mac_addr + 4);
+	UNALIGNED_PUT(sys_rand32_get(), ptr);
+
+	/*
+	 * Clear bit 0 to ensure it isn't a multicast address and set
+	 * bit 1 to indicate address is locally administrered and may
+	 * not be globally unique.
+	 */
+	ctx->mac_addr[0] = (ctx->mac_addr[0] & ~0x01) | 0x02;
+
+	return ctx->mac_addr;
+}
+
+/********************
+ * Radio device API *
+ *******************/
+static enum ieee802154_hw_caps rf2xx_get_capabilities(struct device *dev)
+{
+	struct rf2xx_context *ctx = dev->driver_data;
+	int hwcaps = 0;
+
+	if (ctx->trx_model == RF2XX_TRX_MODEL_212) {
+		hwcaps = IEEE802154_HW_SUB_GHZ;
+	} else {
+		hwcaps = IEEE802154_HW_2_4_GHZ;
+	}
+
+	return IEEE802154_HW_FCS |
+	       IEEE802154_HW_PROMISC |
+	       IEEE802154_HW_FILTER |
+	       IEEE802154_HW_CSMA |
+	       IEEE802154_HW_TX_RX_ACK |
+	       hwcaps;
+}
+
+static int rf2xx_cca(struct device *dev)
+{
+	return -EBUSY;
```

Comment:
so your driver is never going to tx? ( check ieee802154_radio_csma_ca.c )

Reply:
The driver uses transceiver Extended Mode and isn`t recommend execute CCA manually. Droped this method updated driver definition.